src/hoc/withTheme.js

```diff
@@ -23,25 +22,46 @@ const wrapWithTheme = (Component: ReactClass<any>) => {
     static styledComponentId = 'withTheme'
 
     static contextTypes = {
-      [CHANNEL]: PropTypes.func,
+      [CHANNEL]: CONTEXT_CHANNEL_SHAPE,
```

Comment:
@mxstbr that assumes that these people would be using multiple modules/chunks for SC, which is unsupported and broken ^^ But yea, @schwers has an idea for a warning in this case, which puts us on a nice, green path, with flowers and everything 🌹